test/core/simd/simd_f32x4.wast

```diff
@@ -98,129 +61,37 @@
 (assert_return (invoke "f32x4.abs_with_const") (v128.const f32x4 0 1 2 3))
 
 ;; Test different lanes go through different if-then clauses
-;; f32x4.min 0
-(assert_return_canonical_nan
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_0"
-    (v128.const f32x4 nan 0 0 1)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
-  )
-)
-(assert_return_canonical_nan
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_1"
-    (v128.const f32x4 nan 0 0 1)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
-  )
-)
-(assert_return
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_2"
-    (v128.const f32x4 nan 0 0 1)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
-  )
-  (f32.const 0)
-)
+;; f32x4.min
 (assert_return
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_3"
+  (invoke "f32x4.min"
     (v128.const f32x4 nan 0 0 1)
     (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
-  )
-  (f32.const 0)
-)
-;; f32x4.min 1
-(assert_return_canonical_nan
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_0"
-    (v128.const f32x4 nan 0 0 0)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
-  )
-)
-(assert_return_canonical_nan
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_1"
-    (v128.const f32x4 nan 0 0 0)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
-  )
-)
-(assert_return
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_2"
-    (v128.const f32x4 nan 0 0 0)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
   )
-  (f32.const 0)
+  (v128.const f32x4 nan:canonical nan:canonical 0 0)
 )
+;; f32x4.min
 (assert_return
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_3"
+  (invoke "f32x4.min"
     (v128.const f32x4 nan 0 0 0)
     (v128.const f32x4 0 -nan 1 0)
-    (i32.const 0)
-  )
-  (f32.const 0)
-)
-;; f32x4.max 0
-(assert_return_canonical_nan
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_0"
-    (v128.const f32x4 nan 0 0 1)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 1)
-  )
-)
-(assert_return_canonical_nan
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_1"
-    (v128.const f32x4 nan 0 0 1)
-    (v128.const f32x4 0 -nan 1 0)
-    (i32.const 1)
   )
+  (v128.const f32x4 nan:canonical nan:canonical 0 0)
 )
+;; f32x4.max
 (assert_return
-  (invoke "call_indirect_vv_v_f32x4_extract_lane_2"
```

Comment:
Why do we no longer have these extract_lane tests?

Reply:
Before we have unified assert_result* assertion, if we want to test `(invoke "f32x4.min" (v128.const f32x4 nan 0 0 1) (v128.const f32x4 0 -nan 1 0))`, we have to check each lane of the test result which contains partial canonical NAN, that’s why we use `indirect_call` + `extract_lane`.